### PR TITLE
MGMT-11425: Split assisted-baremetal workflow

### PR DESCRIPTION
**Pull request by adriengentil**
Use assisted-common chain:
- In order to not break the existing users of the baremetal workflow, remove `assisted-test-infra-dev-script` image and pull assisted-test-infra sources from the `src` image
- Fix common-gather step using ansible' synchronize instead of fetch, as a consequence, we need to install `rsync` on the baremetal host

Note: I did not replace `assisted-baremetal-test` with `assisted-common-verify` as they do not work in the same way. I can be done later.

### CODE REVIEW

**Comment by adriengentil** [comment]
/override edge-e2e-ai-operator-ztp-ipv4v6-3masters-periodic-ocp-49

**Comment by adriengentil** [comment]
@osherdp @michaellevy101 

I have been a bit "creative", I wanted to avoid to introduce a new image as a dependency (the `assisted-test-infra-dev-script` image) because the workflow is used in many other configs.
Otherwise, that means we would need to add a new base image in all configs and get the approvals against the the repos we don't own [1].

What do you think? (I keep it on hold because I want to run it against more jobs, but I would like to know it the approach is ok before going further)

[1]
```$ grep -rl "assisted-baremetal$" | cut -d'/' -f 4 | sort | uniq -c
      2 assisted-image-service
      2 assisted-installer
      3 assisted-installer-agent
      4 assisted-service
      3 assisted-test-infra
      1 baremetal
      2 cluster-api-provider-agent
     14 cluster-baremetal-operator
     16 cluster-etcd-operator
      6 hypershift
     14 installer
     14 machine-config-operator
      5 release
```

**Comment by adriengentil** [comment]
The gather step is broken, waiting for the gather common step to be merged in order to integrate it https://github.com/openshift/release/pull/28599

**Comment by adriengentil** [comment]
/retitle MGMT-11425: Split assisted-baremetal workflow
/cc @osherdp @eliorerz 
Still, testing but I would be great if you can have a first look :bowing_man:

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (osherdp) [inline ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:157]: is it possible ditching this env?
for example, by copying from podman with relative path to where its at (because I do remember that's his working dir)

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (osherdp) [inline ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:291]: ``podman cp -r`` won't work?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml
Comment (adriengentil) [reply ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml:11]: `assisted-common-verify` doesn't work out of the box, the conformance tests need to be run on from the host, not from prow. I think it can be fixed later.

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:157]: :+1: I will check the workdir

File: ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml
Comment (osherdp) [reply ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml:11]: I agree. It can be fixed later

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:157]:
```
The podman cp command assumes container paths are relative to the container's root directory (/)
```
we need the complete path for `podman cp`

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:291]: there's no `-r` for `podman cp`, but I seems the copy is recursive, I will check :+1:

File: ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml
Comment (adriengentil) [reply ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml:8]: why? There is no chain there, right?

**Comment by adriengentil** [comment]
> All in all, it looks good The only thing I would check closely (except from the jobs passing) is if we're gathering in the same way we do with the existing steps. I remember seeing some weird stuff in vsphere jobs and I'm not sure if it's the fault of the steps or something specific to the vsphere env (and test-infra integration of it)

I fixed the gather step, we get the same things with this change.

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh:157]: hum... we could get the workdir from podman inspect and then build the path, I'll try

File: ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml
Comment (osherdp) [reply ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml:8]: yeah I mean maybe it's better to rename the chain to ``assisted-common-setup`` just to make it symmetric to ``assisted-common-teardown``

Comment (adriengentil) [reply ci-operator/step-registry/assisted/baremetal/assisted-baremetal-workflow.yaml:8]: It makes sense, I think I can do it in a following PR, just to limit the number of changes in this one.

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh]: Looks like it should do the trick :crossed_fingers:

File: ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh
Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/setup/prepare/assisted-common-setup-prepare-commands.sh]: it's much simpler now!

File: ci-operator/step-registry/assisted/common/post/assisted-common-post-commands.sh
Comment (osherdp) [inline ci-operator/step-registry/assisted/common/post/assisted-common-post-commands.sh:22]: which scenario hits it?

Comment (adriengentil) [reply ci-operator/step-registry/assisted/common/post/assisted-common-post-commands.sh:22]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/29138/rehearse-29138-pull-ci-openshift-assisted-test-infra-release-ocm-2.6-e2e-metal-assisted-ha-kube-api/1562718623368220672

Comment (osherdp) [reply ci-operator/step-registry/assisted/common/post/assisted-common-post-commands.sh:22]: It's probably a bug in test-infra not being able to download the kubeconfigs, because those clusters are being installed pretty well
I wouldn't worry about it in the scope of this PR, but we can create a task to fix it

**Comment by adriengentil** [comment]
/test pj-rehearse-max

Last rehearse for sanity, then I think we'll be good to merge.

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
/hold
for waiting or rehearses

**Comment by adriengentil** [comment]
ok it looks good, the failing jobs seem to be unrelated to this change

/unhold

**Pull request merged**
